src/components/fields/location-field/location-modal/location-search/location-search.tsx

```diff
@@ -625,20 +649,21 @@ export const LocationSearch = ({
 						onFocus={handleInputFocus}
 						onChange={handleInputChange}
 						placeholder={addressFieldPlaceholder}
-						readOnly={gettingCurrentLocation}
+						readOnly={gettingCurrentLocation || restrictLocationSelection}
 						value={!gettingCurrentLocation ? queryString : gettingCurrentLocationFetchMessage}
 						ref={inputRef}
 						disabled={disableTextSearch}
```

Comment:
now that we have the need to set the input as readOnly and another prop to disable the text search, may I double confirm both props are needed?

I was under the impression `disableTextSearch` is meant for this but it appears that their behaviours are slightly different, in view of this, what is the use of `disableTextSearch`?

Reply:
updated